In [1]:
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns

In [5]:
df = pd.read_csv('retail_sales_dataset.csv')
df.head()

,Transaction ID,Date,Customer ID,Gender,Age,Product Category,Quantity,Price per Unit,Total Amount
0,1,2023-11-24,CUST001,Male,34,Beauty,3,50,150
1,2,2023-02-27,CUST002,Female,26,Clothing,2,500,1000
2,3,2023-01-13,CUST003,Male,50,Electronics,1,30,30
3,4,2023-05-21,CUST004,Male,37,Clothing,1,500,500
4,5,2023-05-06,CUST005,Male,30,Beauty,2,50,100


In [12]:
df.shape


(1000, 9)

In [9]:
df.dtypes

Transaction ID       int64
Date                object
Customer ID         object
Gender              object
Age                  int64
Product Category    object
Quantity             int64
Price per Unit       int64
Total Amount         int64
dtype: object

In [14]:
df.isnull().sum()

Transaction ID      0
Date                0
Customer ID         0
Gender              0
Age                 0
Product Category    0
Quantity            0
Price per Unit      0
Total Amount        0
dtype: int64

## Initial inspection

The dataset contains 1000 rows and 9 columns. There are no missing 
values in any column. Most columns have appropriate data types 
(int64 for numerical fields), except Date, which is currently stored 
as an object (string) and will need to be converted to datetime 
format for time series analysis.

In [17]:
df.describe()

,Transaction ID,Age,Quantity,Price per Unit,Total Amount
count,1000.000000,1000.00000,1000.000000,1000.000000,1000.000000
mean,500.500000,41.39200,2.514000,179.890000,456.000000
std,288.819436,13.68143,1.132734,189.681356,559.997632
min,1.000000,18.00000,1.000000,25.000000,25.000000
25%,250.750000,29.00000,1.000000,30.000000,60.000000
50%,500.500000,42.00000,3.000000,50.000000,135.000000
75%,750.250000,53.00000,4.000000,300.000000,900.000000
max,1000.000000,64.00000,4.000000,500.000000,2000.000000


In [20]:
for col in ['Age', 'Quantity', 'Price per Unit', 'Total Amount']:
    print(f"{col} mean: {df[col].mean()}")
    print(f"{col} median: {df[col].median()}")
    print(f"{col} mode: {df[col].mode()[0]}")
    print(f"{col} std: {df[col].std()}")
    print("---")

Age mean: 41.392
Age median: 42.0
Age mode: 43
Age std: 13.681429659122527
---
Quantity mean: 2.514
Quantity median: 3.0
Quantity mode: 4
Quantity std: 1.1327343409145356
---
Price per Unit mean: 179.89
Price per Unit median: 50.0
Price per Unit mode: 50
Price per Unit std: 189.68135627129232
---
Total Amount mean: 456.0
Total Amount median: 135.0
Total Amount mode: 50
Total Amount std: 559.997631555123
---


## Descriptive Statistics

The average customer age is 41.4 years (median 42, mode 43), showing 
a fairly even age distribution. For Total Amount, the mean (456.0) is 
much higher than the median (135.0), indicating that a small number of 
high-value transactions are pulling the average up — most purchases 
are actually smaller. The same pattern appears in Price per Unit 
(mean 179.89 vs median 50.0), suggesting a mix of low-cost and 
occasional high-cost products in the store's catalog.